src/validator.spec.ts

```diff
@@ -59,7 +61,194 @@ describe('fonk-array-required-validator specs', () => {
     expect(result).toEqual({
       succeeded: false,
       message: 'other message',
-      type: '',
+      type: VALIDATOR_TYPE,
+    });
+  });
+
+  it('should return failed validation when type of feeds value is string', () => {
+    // Arrange
+    const value = 'test';
+
+    // Act
+    const result = validator({
+      value,
+    });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: DEFAULT_MESSAGE,
+      type: VALIDATOR_TYPE,
+    });
+  });
+
+  it('should return failed validation when it feeds value is true', () => {
+    // Arrange
+    const value = true;
+
+    // Act
+    const result = validator({
+      value,
+    });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: DEFAULT_MESSAGE,
+      type: VALIDATOR_TYPE,
+    });
+  });
+
+  it('should return failed validation when it feeds value is false', () => {
+    // Arrange
+    const value = false;
+
+    // Act
+    const result = validator({
+      value,
+    });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: DEFAULT_MESSAGE,
+      type: VALIDATOR_TYPE,
+    });
+  });
+
+  it('should return failed validation when it feeds value is an object', () => {
+    // Arrange
+    const value = {};
+
+    // Act
+    const result = validator({
+      value,
+    });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: DEFAULT_MESSAGE,
+      type: VALIDATOR_TYPE,
+    });
+  });
+
+  it('should return failed validation when it feeds value is a function', () => {
+    // Arrange
+    const value = () => null;
+
+    // Act
+    const result = validator({
+      value,
+    });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: DEFAULT_MESSAGE,
+      type: VALIDATOR_TYPE,
+    });
+  });
+
+  it('should return succeeded validation when it feeds value is an array', () => {
+    // Arrange
+    const value = [];
+
+    // Act
+    const result = validator({
+      value,
+    });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: true,
```

Comment:
This spec should be fail. Because the main goal of this validator is to have an array with items

Reply:
Done!